ui/src/TimeOnProductPage/TimeOnProduct.tsx

```diff
@@ -77,29 +76,24 @@ export const sortTimeOnProductItems = (a: TimeOnProductItem, b: TimeOnProductIte
     return returnValue;
 };
 
-export interface TimeOnProductProps {
-    currentSpace: Space;
-}
-
-function TimeOnProduct({currentSpace}: TimeOnProductProps): JSX.Element {
+function TimeOnProduct(): JSX.Element {
     const { teamUUID = '' } = useParams<{ teamUUID: string }>();
-    const navigate = useNavigate();
 
     const [modalContents, setModalContents] = useRecoilState(ModalContentsState)
     const viewingDate = useRecoilValue(ViewingDateState);
     const isReadOnly = useRecoilValue(IsReadOnlyState);
 
-    const { fetchProducts, products } = useFetchProducts();
+    const { fetchProducts, products } = useFetchProducts(teamUUID);
+    const { fetchCurrentSpace, currentSpace } = useFetchCurrentSpace(teamUUID);
 
     const [isLoading, setIsLoading] = useState<boolean>(false);
 
     useEffect(() => {
-        // @todo fetch space if no space is present
-        if (!currentSpace) navigate(`/${teamUUID}`);
-    }, [currentSpace, navigate, teamUUID]);
+        if (!currentSpace.uuid) fetchCurrentSpace();
```

Comment:
Changed it so that instead of navigating to the space page, it just fetches the current space if its not already fetched. Thus allowing you to go directly to the time on product page.